src/methods.cc

```diff
@@ -3,29 +3,46 @@
 ** Licensed under MIT License.
 */
 
+v8::Local<v8::Value> Get(v8::Local<v8::Object> object, v8::Local<v8::Value> key) {
+#if NODE_MODULE_VERSION >= NODE_4_0_MODULE_VERSION
+  return object->Get(object->CreationContext(), key).ToLocalChecked();
+#else
+  return object->Get(key);
+#endif
+}
+
+void Set(v8::Local<v8::Object> object, v8::Local<v8::Value> key, v8::Local<v8::Value> value) {
+#if NODE_MODULE_VERSION >= NODE_4_0_MODULE_VERSION
+  (void) object->Set(object->CreationContext(), key, value);
+#else
+  object->Set(key, value);
+#endif
+}
+
 void FSEvents::emitEvent(const char *path, UInt32 flags, UInt64 id) {
-  if (!handler) return;
   Nan::HandleScope handle_scope;
+  v8::Local<v8::Object> object = handle();
+  Nan::Callback handler(Get(object, Nan::New<v8::String>("handler").ToLocalChecked()).As<v8::Function>());
```

Comment:
Can you use `Nan::To<v8::Function>(...)`?  It does an `IsFunction()` check before casting.